### PR TITLE
Separate formation notifications

### PR DESCRIPTION
**Pull request by ivantenevvasilev**
**Description**
The unit tests for assign formation and unassign formation have become too big and everytime we have to make changes to the notifications or assignment/unassignment of formations, we have to deal with the enormous unit test files. This PR aims to separate them into two independent services so it is easier to maintain and test them.

Changes proposed in this pull request:
- move the generation and sending of notifications in its own service

**Related issue(s)**

- ...

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [x] [N/A] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [x] Mocks are regenerated, using the automated script

### CODE REVIEW

File: components/director/internal/domain/formation/notifications.go [hunk 1]
```diff
@@ -0,0 +1,573 @@
...
+		return nil, errors.Wrapf(err, "while getting runtime labels for id %s", runtimeID)
+	}
+	runtimeWithLabels := &webhookdir.RuntimeWithLabels{
+		Runtime: runtime,
+		Labels:  runtimeLabels,
+	}
+
+	webhook, err := ns.webhookRepository.GetByIDAndWebhookType(ctx, tenant, runtimeID, model.RuntimeWebhookReference, model.WebhookTypeConfigurationChanged)
+	if err != nil {
+		if apperrors.IsNotFoundError(err) {
```
Comment (nyordanoff) [inline components/director/internal/domain/formation/notifications.go:464]: Maybe we can add unit test for that error case

File: components/director/internal/domain/formation/notifications.go [hunk 2]
```diff
@@ -0,0 +1,573 @@
...
+
+	log.C(ctx).Infof("Total number of app-to-app notifications for application with ID %s that is being %s is %d", appID, operation, len(requests))
+
+	return requests, nil
+}
+
+func (ns *notificationsService) generateRuntimeNotificationsForRuntimeContextAssignment(ctx context.Context, tenant, runtimeCtxID string, formation *model.Formation, operation model.FormationOperation) ([]*webhookclient.Request, error) {
+	log.C(ctx).Infof("Generating %s notifications for runtime context %s", operation, runtimeCtxID)
+	runtimeCtx, err := ns.runtimeContextRepo.GetByID(ctx, tenant, runtimeCtxID)
+	if err != nil {
```
Comment (nyordanoff) [inline components/director/internal/domain/formation/notifications.go:424]: Maybe we can add unit test for that error case

File: components/director/cmd/director/main.go [hunk 3]
```diff
@@ -654,7 +654,8 @@ func runtimeSvc(cfg config, securedHTTPClient, mtlsHTTPClient, extSvcMtlsHTTPCli
 	asaSvc := scenarioassignment.NewService(asaRepo, labelDefinitionSvc)
 	tenantSvc := tenant.NewServiceWithLabels(tenantRepo, uidSvc, labelRepo, labelSvc)
 	webhookClient := webhookclient.NewClient(securedHTTPClient, mtlsHTTPClient, extSvcMtlsHTTPClient)
-	formationSvc := formation.NewService(labelDefinitionRepo, labelRepo, formationRepo, formationTemplateRepo, labelSvc, uidSvc, labelDefinitionSvc, asaRepo, asaSvc, tenantSvc, runtimeRepo, runtimeContextRepo, webhookRepo, webhookClient, appRepo, appTemplateRepo, webhookConverter, cfg.Features.RuntimeTypeLabelKey, cfg.Features.ApplicationTypeLabelKey)
+	notificationsService := formation.NewNotificationService(appRepo, appTemplateRepo, runtimeRepo, runtimeContextRepo, labelRepo, webhookRepo, webhookConverter, webhookClient)
```
Comment (nyordanoff) [inline components/director/cmd/director/main.go:657]: super minor: across all changed `main.go` `root_resolver` and `builder.go` files, we can rename `notificationsService` to `notificationsSvc` just for consistency purposes

**Review by dzahariev — APPROVED** [review]

**Review by nyordanoff — APPROVED** [review]

**Review by StefanCholakov — APPROVED** [review]

**Pull request merged**